### PR TITLE
Feature/id 02/user management

### PR DESCRIPTION
**Pull request by leonardo-herrera-jalafund**
User management:
User service & User controller implemented

### CODE REVIEW

File: src/Presentation/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        if(await _userService.GetUserByIdAsync(id) == null)
+        {
+            return NotFound();
+        }
+
+        await _userService.DeleteUserAsync(id);
+        return Ok();
+    }
+}
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs]: Missing endline

File: src/Presentation/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    public async Task<IActionResult> GetUserById(Guid id)
+    {
+        var user = await _userService.GetUserByIdAsync(id);
+
+        if(user == null)
+        {
+            return NotFound();
+        }
+
+        return Ok();
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs]: The return should be user and not only "ok" assuming we have an admin view.

File: src/Presentation/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace ExpenseTrackerGrupo4.src.Presentation.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class UserController(IUserService userService) : ControllerBase
+{
+    private readonly IUserService _userService = userService;
+
+    [HttpGet("{id}")]
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs:19]: missing authorization

File: src/Presentation/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        if(user == null)
+        {
+            return NotFound();
+        }
+
+        return Ok();
+    }
+
+    [HttpPut]
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs:32]: Missing authorization

File: src/Presentation/Controllers/UserController.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    public async Task<IActionResult> GetUserByEmail(string email)
+    {
+        var user = await _userService.GetUserByEmailAsync(email);
+
+        if(user == null)
+        {
+            return NotFound();
+        }
+
+        return Ok();
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs]: The return should be user and not only "ok" assuming we have an admin view.

File: src/Presentation/Controllers/UserController.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        if(user == null)
+        {
+            return BadRequest();
+        }
+        
+        await _userService.UpdateUserAsync(user);
+        return Ok();
+    }
+
+    [HttpGet("email/{email}")]
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs:72]: Missing athorization

File: src/Presentation/Controllers/UserController.cs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    [HttpPut]
+    public async Task<IActionResult> UpdateUser([FromBody] User user)
+    {
+        if(user == null)
+        {
+            return BadRequest();
+        }
+        
+        await _userService.UpdateUserAsync(user);
+        return Ok();
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs]: When updating the data, the password is not hashed again.
image

File: src/Presentation/Controllers/UserController.cs [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+        if(user == null)
+        {
+            return NotFound();
+        }
+
+        return Ok();
+    }
+
+    [HttpDelete("{id}")]
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs:85]: missing authorization

**Review by luis-espinoza-jalafund — CHANGES_REQUESTED** [review]

**Review by 0santisimo0 — COMMENTED** [review]
Implement IRepository to IUserRepository

**Review by 0santisimo0 — CHANGES_REQUESTED** [review]
Implement IRepository to IUserRepository

**Review by santiago-caballero-jalafund — CHANGES_REQUESTED** [review]
Implement IRepository to IUserRepository (See The UML Diagram)

**Review by santiago-caballero-jalafund — CHANGES_REQUESTED** [review]
Use a DTO for User to not expose the id

**Review by 0santisimo0 — APPROVED** [review]

File: src/Presentation/Controllers/UserController.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        if(user == null)
+        {
+            return NotFound(new { Message = "User not found." });
+        }
+
+        return Ok(user);
+    }
+
+    [HttpPut]
+    public async Task<IActionResult> UpdateUser([FromBody] UserUpdateRequestDTO userUpdated)
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Controllers/UserController.cs]: The id should not be requested in the body, it should be a field like getById or Delete

image

File: src/Presentation/DTOs/UserUpdateRequestDTO.cs [hunk 10]
```diff
@@ -0,0 +1,8 @@
+namespace ExpenseTrackerGrupo4.src.Presentation.DTOs;
+
+public class UserUpdateRequestDTO
+{
+    public required Guid Id { get; set; }
+    public required string Name { get; set; }
+    public required string Email { get; set; }
+}
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/DTOs/UserUpdateRequestDTO.cs]: Missing endline

File: src/Presentation/Profiles/UserProfile.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+using ExpenseTrackerGrupo4.src.Presentation.DTOs;
+
+public class UserProfile : Profile
+{
+    public UserProfile()
+    {
+        CreateMap<UserUpdateRequestDTO, User>()
+            .ForMember(dest => dest.PasswordHash, opt => opt.Ignore());
+    }
+}
```
Comment (luis-espinoza-jalafund) [inline src/Presentation/Profiles/UserProfile.cs]: Missing endline

**Review by luis-espinoza-jalafund — CHANGES_REQUESTED** [review]

**Review by santiago-caballero-jalafund — APPROVED** [review]

**Review by luis-espinoza-jalafund — APPROVED** [review]

**Pull request merged**